pkg/rehearse/jobs.go

```diff
@@ -408,6 +408,77 @@ func AddRandomJobsForChangedTemplates(templates []config.ConfigMapSource, toBeRe
 	return rehearsals
 }
 
+func getPresubmitForRegistryStep(node registry.Node, configs config.ByFilename, prConfigPresubmits map[string][]prowconfig.Presubmit) (string, *prowconfig.Presubmit) {
+	// get sorted list of configs keys to make the function deterministic
+	var keys []string
+	for k := range configs {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		ciopConfig := configs[key]
+		tests := ciopConfig.Configuration.Tests
+		repoPresubmits := prConfigPresubmits[ciopConfig.Info.Repo]
+		for _, test := range tests {
+			if test.MultiStageTestConfiguration != nil {
```

Comment:
I think we can make this a bit more readable by inverting this into a guard clause, allowing to de-indent the rest of the block.